From 84a312dd5d0ed9fb8dade3a703dda2c7e6754870 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:54:37 -0400
Subject: [fs] nfs: populate ->net in mount data when remounting

Message-id: <1403798077-28799-79-git-send-email-steved@redhat.com>
Patchwork-id: 84943
O-Subject: [RHEL7.1 PATCH 78/78] NFS: populate ->net in mount data when remounting
Bugzilla: 1113248
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Mateusz Guzik <mguzik@redhat.com>

Otherwise the kernel oopses when remounting with IPv6 server because
net is dereferenced in dev_get_by_name.

Use net ns of current thread so that dev_get_by_name does not operate on
foreign ns. Changing the address is prohibited anyway so this should not
affect anything.

Signed-off-by: Mateusz Guzik <mguzik@redhat.com>
Cc: linux-nfs@vger.kernel.org
Cc: linux-kernel@vger.kernel.org
Cc: stable@vger.kernel.org # 3.4+
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: a914722f333b3359d2f4f12919380a334176bb89
Bugzilla: 1113248
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/super.c b/fs/nfs/super.c
index 6766eba..e08eba9 100644
--- a/fs/nfs/super.c
+++ b/fs/nfs/super.c
@@ -2257,6 +2257,7 @@ nfs_remount(struct super_block *sb, int *flags, char *raw_data)
  data->nfs_server.addrlen = nfss->nfs_client->cl_addrlen;
  data->version = nfsvers;
  data->minorversion = nfss->nfs_client->cl_minorversion;
+ data->net = current->nsproxy->net_ns;
  memcpy(&data->nfs_server.address, &nfss->nfs_client->cl_addr,
   data->nfs_server.addrlen);
 
-- 
1.7.1